bundles/org.openhab.binding.smarther/src/main/java/org/openhab/binding/smarther/internal/handler/SmartherModuleHandler.java

```diff
@@ -0,0 +1,733 @@
+                            }
+                        }
+                        this.chronothermostat = statusChrono;
+                        logger.debug("Module[{}] Status: [{}]", thing.getUID(), this.chronothermostat);
+                    } else {
+                        throw new SmartherGatewayException("No chronothermostat data found");
+                    }
+
+                    // Refresh the programs list for "automatic" mode
+                    final ExpiringCache<List<Program>> localProgramCache = this.programCache;
+                    if (localProgramCache != null) {
+                        final List<Program> programs = localProgramCache.getValue();
+                        if (programs != null) {
+                            dynamicStateDescriptionProvider.setPrograms(programChannelUID, programs);
+                        }
+                    }
+
+                    updateModuleStatus();
+
+                    getBridgeHandler().registerNotification(config.getPlantId());
+
+                    // Everything is ok > set the Thing state to Online
+                    updateStatus(ThingStatus.ONLINE);
+                    return true;
+                } else if (thing.getStatus() != ThingStatus.OFFLINE) {
+                    logger.debug("Module[{}] Switched {} as Bridge is not online", thing.getUID(),
+                            bridgeStatusInfo.getStatus());
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Smarther Bridge Offline");
+                }
+            }
+            return false;
+        } catch (SmartherIllegalPropertyValueException e) {
+            logger.debug("Module[{}] Illegal property value error during polling: {}", thing.getUID(), e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
+            schedulePoll();
+            return false;
+        } catch (SmartherSubscriptionAlreadyExistsException e) {
+            logger.debug("Module[{}] Subscription error during polling: {}", thing.getUID(), e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
+            schedulePoll();
+            return false;
+        } catch (SmartherGatewayException e) {
+            logger.warn("Module[{}] API Gateway error during polling: {}", thing.getUID(), e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            schedulePoll();
+            return false;
+        } catch (RuntimeException e) {
+            // All other exceptions apart from Subscription and Gateway issues
+            logger.warn("Module[{}] Unexpected error during polling, please report if this keeps occurring: ",
+                    thing.getUID(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
+            schedulePoll();
+            return false;
```

Comment:
might as well put these two statements to after the catch blocks since all the catch blocks have these two statements.

Reply:
I agree it would be better